### PR TITLE
2 improvements for Solaris platform:

### PR DESCRIPTION
**Pull request by rbrunckhorst**
- Solaris workaround with kadmind startup problem
  - Improvement for Developer Studio on Solaris

### CODE REVIEW

**Review by meffie — APPROVED** [review]
Interesting. Thank you.

File: roles/openafs_krbserver/templates/krb5.conf.j2 [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 [libdefaults]
     default_realm = {{ afs_realm }}
+    dns_lookup_kdc = false
```
Comment (meffie) [inline roles/openafs_krbserver/templates/krb5.conf.j2]: Yes, but in some cases we want to lookup the kdc with dns. For example when running on linux on real cells, we use srv records to lookup the kdcs. Perhaps this should be a platform specific value?  Or am I mis-understanding?

**Review by meffie — CHANGES_REQUESTED** [review]
Is this really platform specific? if so i'd feel more comfortable if we added a platform specific variable to roles/openafs_krbserver/vars/*.yml files.

**Comment by rbrunckhorst** [comment]
This was a workaround on Solaris to start kadmind when there is no real 
DNS-server setup in place.
I have tested that as well for other platforms without any problems.

But for real cells it should be not included.
Maybe it  could be enabled by a variable in the molecule file:
   afs_kdcs_dns_lookup: 'true'|false (default: true)

[roles/openafs\_krbserver/templates/krb5.conf.j2]:
[libdefaults]
    default_realm = {{ afs_realm }}
    dns_lookup_kdc = {{ afs_kdcs_dns_lookup }}

roles/openafs_krbserver/vars/*.yml:
   afs_kdcs_dns_lookup: 'false' (Solaris)
   afs_kdcs_dns_lookup: 'true' (all others)

Maybe also for krbclient ?

**Comment by meffie** [comment]
On Wed, 29 Dec 2021 09:08:11 -0800
Ralf Brunckhorst ***@***.***> wrote:

> This was a workaround on Solaris to start kadmind when there is no real 
> DNS-server setup in place.
> I have tested that as well for other platforms without any problems.
> 
> But for real cells it should be not included.

Yes, keep in mind, the OpenAFS Ansible roles are used for real cells as
well as test cells.

> Maybe it  could be enabled by a variable in the molecule file:
>    afs_kdcs_dns_lookup: 'true'|false (default: true)

Yes, and for a playbook for real cell, you would provide this
variable as an inventory variable.

 
> [roles/openafs\_krbserver/templates/krb5.conf.j2]:
> [libdefaults]
>     default_realm = {{ afs_realm }}
>     dns_lookup_kdc = {{ afs_kdcs_dns_lookup }}
> 
> roles/openafs_krbserver/vars/*.yml:
>    afs_kdcs_dns_lookup: 'false' (Solaris)
>    afs_kdcs_dns_lookup: 'true' (all others)

Yes, that seems to be better. I think the other variables are prefixed with
'afs_krb_', so for this a better name would be 'afs_krb_dns_lookup_kdc'

> Maybe also for krbclient ?

Yes, the same would need to be done for the client role. I think the prefix
we used there is 'afs_krbclient_'.

Thank you.

-- 
Michael Meffie ***@***.***>

**Comment by rbrunckhorst** [comment]
Hi Mike,

I think I have found a better workaround:
git diff                                                                 
                                                                                                                                                       ─╯
diff --git a/roles/openafs_krbserver/templates/krb5.conf.j2 
b/roles/openafs_krbserver/templates/krb5.conf.j2
index f0d4aa8..bad2f27 100644
--- a/roles/openafs_krbserver/templates/krb5.conf.j2
+++ b/roles/openafs_krbserver/templates/krb5.conf.j2
@@ -2,10 +2,10 @@

  [libdefaults]
      default_realm = {{ afs_realm }}
-    dns_lookup_kdc = false

  [realms]
      {{ afs_realm }} = {
          kdc = {{ ansible_hostname }}
          admin_server = {{ ansible_hostname }}
+        kpasswd_server = {{ ansible_hostname }}
      }

So dns_lookup_kdc is not needed anymore, just adding a line 
kpasswd_server.
 From the man-page of krb5.conf:
kpasswd_server
               Points to the server where all the password changes are 
performed.  If there is no such entry, DNS will be queried (unless 
forbidden by dns_lookup_kdc).  Finally, port 464 on the admin_server 
host will be tried.

I'm not 100% sure if that is the root-cause because I would assume that 
it will only use the setting if running kapsswd, but with that kadmind 
is starting fine on my Solaris box.

What do you think?

Should I create a new PR?
Also for the client part?

/Ralf

On 29 Dec 2021, at 19:56, Michael Meffie wrote:

> On Wed, 29 Dec 2021 09:08:11 -0800
> Ralf Brunckhorst ***@***.***> wrote:
>
>> This was a workaround on Solaris to start kadmind when there is no 
>> real
>> DNS-server setup in place.
>> I have tested that as well for other platforms without any problems.
>>
>> But for real cells it should be not included.
>
> Yes, keep in mind, the OpenAFS Ansible roles are used for real cells 
> as
> well as test cells.
>
>> Maybe it  could be enabled by a variable in the molecule file:
>>    afs_kdcs_dns_lookup: 'true'|false (default: true)
>
> Yes, and for a playbook for real cell, you would provide this
> variable as an inventory variable.
>
>
>> [roles/openafs\_krbserver/templates/krb5.conf.j2]:
>> [libdefaults]
>>     default_realm = {{ afs_realm }}
>>     dns_lookup_kdc = {{ afs_kdcs_dns_lookup }}
>>
>> roles/openafs_krbserver/vars/*.yml:
>>    afs_kdcs_dns_lookup: 'false' (Solaris)
>>    afs_kdcs_dns_lookup: 'true' (all others)
>
> Yes, that seems to be better. I think the other variables are prefixed 
> with
> 'afs_krb_', so for this a better name would be 
> 'afs_krb_dns_lookup_kdc'
>
>> Maybe also for krbclient ?
>
> Yes, the same would need to be done for the client role. I think the 
> prefix
> we used there is 'afs_krbclient_'.
>
> Thank you.
>
> -- 
> Michael Meffie ***@***.***>
>
>
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/openafs-contrib/ansible-openafs/pull/51#issuecomment-1002734676
> You are receiving this because you authored the thread.
>
> Message ID: 
> ***@***.***>

**Comment by meffie** [comment]
On Thu, 30 Dec 2021 03:55:32 -0800
Ralf Brunckhorst ***@***.***> wrote:

> Hi Mike,
> 
> I think I have found a better workaround:
> git diff                                                                 
>                                                                                                                                                        ─╯
> diff --git a/roles/openafs_krbserver/templates/krb5.conf.j2 
> b/roles/openafs_krbserver/templates/krb5.conf.j2
> index f0d4aa8..bad2f27 100644
> --- a/roles/openafs_krbserver/templates/krb5.conf.j2
> +++ b/roles/openafs_krbserver/templates/krb5.conf.j2
> @@ -2,10 +2,10 @@
> 
>   [libdefaults]
>       default_realm = {{ afs_realm }}
> -    dns_lookup_kdc = false
> 
>   [realms]
>       {{ afs_realm }} = {
>           kdc = {{ ansible_hostname }}
>           admin_server = {{ ansible_hostname }}
> +        kpasswd_server = {{ ansible_hostname }}
>       }
> 
> So dns_lookup_kdc is not needed anymore, just adding a line 
> kpasswd_server.
>  From the man-page of krb5.conf:
> kpasswd_server
>                Points to the server where all the password changes are 
> performed.  If there is no such entry, DNS will be queried (unless 
> forbidden by dns_lookup_kdc).  Finally, port 464 on the admin_server 
> host will be tried.
> 
> I'm not 100% sure if that is the root-cause because I would assume that 
> it will only use the setting if running kapsswd, but with that kadmind 
> is starting fine on my Solaris box.
> 
> What do you think?

Thanks Ralf. Let me take a closer look.

> 
> Should I create a new PR?
> Also for the client part?

It might be helpful to update this PR (#51) to just include the
changes needed for solaris studio (aka developer studio), and make
a new PR for change to the kerberos client and server roles.

-- 
Michael Meffie ***@***.***>

**Comment by rbrunckhorst** [comment]
I will close the PR and will create 2 new PR.

**Pull request closed without merging**